bundles/binding/org.openhab.binding.denon/src/main/java/org/openhab/binding/denon/internal/DenonBinding.java

```diff
@@ -0,0 +1,244 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.denon.internal;
+
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.openhab.binding.denon.DenonBindingProvider;
+import org.openhab.core.binding.AbstractActiveBinding;
+import org.openhab.core.binding.BindingProvider;
+import org.openhab.core.items.ItemNotFoundException;
+import org.openhab.core.items.ItemRegistry;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+	
+/**
+ * Binding that communicates with one or multiple Denon receivers. 
+ * 
+ * @author Jeroen Idserda
+ * @since 1.7.0
+ */
+public class DenonBinding extends AbstractActiveBinding<DenonBindingProvider> implements ManagedService {
+
+	private static final Logger logger = LoggerFactory.getLogger(DenonBinding.class);
+	
+	private Map<String, DenonConnectionProperties> connections = new HashMap<String, DenonConnectionProperties>();
+	
+	private int refreshInterval = 5000;
+
+	public void deactivate() {
+		for (Entry<String, DenonConnectionProperties> entry : connections.entrySet()) {
+			entry.getValue().getConnector().disconnect();
+		}
+	}
+
+	@Override
+	protected void execute() {
+		for (Entry<String, DenonConnectionProperties> entry : connections.entrySet()) {
+			DenonConnectionProperties connection = entry.getValue();
+			if (connection.isHttp()) {
+				entry.getValue().getConnector().updateState();
+			}
+		}
+	}
+
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	@Override
+	protected String getName() {
+		return "Denon Refresh Service";
+	}
+	
+	/**
+	 * @{inheritDoc}
+	 */	
+	@Override
+	public void bindingChanged(BindingProvider provider, String itemName) {
+		logger.debug("Denon binding changed for item {}", itemName);
+		if (provider instanceof DenonBindingProvider)
+		{	
+			DenonBindingConfig config = ((DenonBindingProvider) provider).getConfig(itemName);
+			if (config != null) {
+				connections.get(config.getInstance()).getConnector().updateStateFromCache(config.getProperty());
```

Comment:
since the phrase ```connections.get(config.getInstance()).getConnector()``` is duplicated a lot it might be worth refactoring it into a new method?